pkg/sharings/sharings.go

```diff
@@ -204,8 +202,39 @@ func ShareDoc(instance *instance.Instance, sharing *Sharing, recStatus *Recipien
 			}
 		}
 
+		var values []string
+
+		// Dynamic sharing
+		if rule.Selector != "" {
+			// Create index based on selector to retrieve documents to share
+			index := mango.IndexOnFields(docType, sharing.ID(), []string{rule.Selector})
```

Comment:
Right, but I guess I have to handle the case where the index already exists, which will return error.
Or maybe there is a way to query the existing indexes ?